### PR TITLE
solution to problem 8

### PR DESCRIPTION
**Pull request by pxa4576**

### CODE REVIEW

File: curriculum/js1/8.js [hunk 1]
```diff
@@ -5,8 +5,15 @@
  * @returns {number}
  */
 
-const solution = (a, b) => {
-  return 0;
+const solution = (a,b,gcd=1,div=1) => {
+  if(a==0 || b==0)
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Please use `===`

File: curriculum/js1/8.js [hunk 2]
```diff
@@ -5,8 +5,15 @@
...
+    return;
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Please combine this line with line 9

File: curriculum/js1/8.js [hunk 3]
```diff
@@ -5,8 +5,15 @@
...
+if(a%div==0 && b%div==0)
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Please use `===`

File: curriculum/js1/8.js [hunk 4]
```diff
@@ -5,8 +5,15 @@
...
+   x=div;
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Try writing this without `x`

File: curriculum/js1/8.js [hunk 5]
```diff
@@ -5,8 +5,15 @@
...
+if(div >=a || div >=b)
+  return gcd = x;
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: `return gcd;`

File: curriculum/js1/8.js [hunk 6]
```diff
@@ -5,8 +5,11 @@
  * @returns {number}
  */
 
-const solution = (a, b) => {
-  return 0;
+const solution = (a,b,gcd=1,div=1) => {
+  if (div>a || div>b) return gcd;
+   if (a%div===0 && b%div===0) gcd=div;
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Please review indentation on this line. It appears to be out of alignment.

File: curriculum/js1/8.js [hunk 7]
```diff
@@ -5,8 +5,11 @@
...
+return solution(a,b,gcd,div+1);
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Please review indentation on this line. It appears to be out of alignment.

File: curriculum/js1/8.js [hunk 8]
```diff
@@ -5,8 +5,11 @@
...
+
```
Comment (danielthalhuber) [inline curriculum/js1/8.js]: Minor, general comment on horizontal spacing: most code styles will recommend a horizontal space after commas and around equal signs to make the code easier to read.

E.g.:
```javascript
solution(a, b, gcd, div+1)
```
```javascript
if (a%div === 0 && b%div === 0) gcd = div;
```

Comment (pxa4576) [reply curriculum/js1/8.js]: sure ill fix it.

**Review by seemcat — APPROVED** [review]

**Pull request merged**